src/pages/task/taskTabs/TestsTable.tsx

```diff
@@ -17,92 +23,56 @@ import {
   TaskQuery,
 } from "gql/generated/types";
 import { TASK_TESTS } from "gql/queries";
+import { useTableSort, useUpdateURLQueryParams, usePolling } from "hooks";
+import { useQueryParams } from "hooks/useQueryParam";
 import {
-  useUpdateURLQueryParams,
-  usePolling,
-  useStatusesFilter,
-  useFilterInputChangeHandler,
-} from "hooks";
-import { RequiredQueryParams, TableOnChange } from "types/task";
+  RequiredQueryParams,
+  mapFilterParamToId,
+  mapIdToFilterParam,
+} from "types/task";
 import { TestStatus } from "types/test";
 import { queryString, url } from "utils";
 import { getColumnsTemplate } from "./testsTable/getColumnsTemplate";
 
 const { getLimitFromSearch, getPageFromSearch } = url;
 const { parseQueryString, queryParamAsNumber } = queryString;
+const { getDefaultOptions: getDefaultFiltering } = Filters;
+const { getDefaultOptions: getDefaultSorting } = Sorting;
 
 interface TestsTableProps {
   task: TaskQuery["task"];
 }
+
 export const TestsTable: React.FC<TestsTableProps> = ({ task }) => {
   const { pathname, search } = useLocation();
   const updateQueryParams = useUpdateURLQueryParams();
-  const taskAnalytics = useTaskAnalytics();
-  const sendFilterTestsEvent = (filterBy: string) =>
-    taskAnalytics.sendEvent({ name: "Filter Tests", filterBy });
+  const { sendEvent } = useTaskAnalytics();
 
   const queryVariables = getQueryVariables(search, task.id);
-  const { limitNum, pageNum, sort } = queryVariables;
+  const [queryParams, setQueryParams] = useQueryParams();
+  const { execution, limitNum, pageNum, sort } = queryVariables;
   const cat = sort?.[0]?.sortBy;
-  const dir = sort?.[0]?.direction;
 
   const appliedDefaultSort = useRef(null);
   useEffect(() => {
+    // Avoid race condition where this hook overwrites TaskTabs setting a default execution.
+    if (execution == null) {
+      return;
+    }
+
     if (
       cat === undefined &&
       updateQueryParams &&
       appliedDefaultSort.current !== pathname
     ) {
       appliedDefaultSort.current = pathname;
       updateQueryParams({
-        [RequiredQueryParams.Category]: TestSortCategory.Status,
-        [RequiredQueryParams.Sort]: SortDirection.Asc,
+        [TableQueryParams.SortBy]: TestSortCategory.Status,
+        [TableQueryParams.SortDir]: SortDirection.Asc,
```

Comment:
It's also possible to set execution here using the execution from the `task` prop, not sure if that's a good option though, just wanted to mention it

Reply:
Since all of the task tabs require an execution I think it's preferable to include it in a higher level shared component!